frontend/src/services/api/polls.ts

```diff
@@ -1,40 +1,104 @@
 import { Poll } from "../../models/polls";
 import { AxiosResponse } from "axios";
-import { apiService, unwrap } from "../apiService";
+import { apiService, PaginatedResponse, unwrap } from "../apiService";
+import { sortingToApiParameter } from "../../components/utils/table/sorting";
+import {
+    PollStateFilter,
+    pollStateFilterToApiParameter
+} from "../../components/polls/polls_table/PollsTableFilter";
+import { joinNonEmpty } from "../../utils/parameter";
 
-export const polls = {
-    list: () =>
-        unwrap<Poll[]>(
-            apiService
-                .get<Poll[]>("/polls/")
-                .then((response: AxiosResponse<Poll[]>) => {
-                    // Parse the date (because it's not a datetime).
-                    response.data.forEach(
-                        poll =>
-                            (poll.publicationDate = poll.publicationDate
-                                ? new Date(poll.publicationDate)
-                                : undefined)
-                    );
+/**
+ * Parse the `publicationDate` and `creationDateTime` JSON field.
+ *
+ * Should be called in a `then` after an `apiService.get("/polls/...")`.
+ */
+const parseDates = (response: AxiosResponse<PaginatedResponse<Poll[]>>) => {
+    response.data.results.forEach(poll => {
+        poll.creationDateTime = new Date(poll.creationDateTime);
+        poll.publicationDate = poll.publicationDate
+            ? new Date(poll.publicationDate)
+            : undefined;
+    });
+
+    return response;
+};
 
-                    return response;
-                })
-        ),
+const listGeneric = (parameters: string = "") => (page = 1) =>
+    unwrap<PaginatedResponse<Poll[]>>(
+        apiService
+            .get<PaginatedResponse<Poll[]>>(
+                `/polls/?page=${page}${
+                    parameters === undefined ? "" : "&"
+                }${parameters}`
+            )
+            .then(parseDates)
```

Comment:
Does this return a promise?

Reply:
It returns an `AxiosResponse`, like `apiService.get`

Reply:
`apiService.*.get` returns a `Promise<AxiosResponse<_>>`